src/nrniv/CMakeLists.txt

```diff
@@ -322,3 +322,55 @@ endif()
 # ~~~
 install(TARGETS nrniv nocmodl modlunit DESTINATION bin)
 install(TARGETS nrniv_lib DESTINATION ${NRN_INSTALL_SHARE_LIB_DIR})
+if(MINGW)
+  set(mswindir ${PROJECT_SOURCE_DIR}/src/mswin)
+  add_custom_command(OUTPUT nrnicon.res
+    COMMAND echo "id ICON \"${mswindir}/nrniv10.ico\"" > nrnicon.rc
+    COMMAND windres nrnicon.rc -O coff -o nrnicon.res
+    DEPENDS ${mswindir}/nrniv10.ico)
+  set_source_files_properties(
+    ${mswindir}/extra/neuron.cpp
+    ${mswindir}/extra/mos2nrn.cpp
+    ${mswindir}/extra/nrnbinstr.cpp
+    PROPERTIES COMPILE_DEFINITIONS MINGW)
+  add_executable(neuron ${mswindir}/extra/neuron.cpp nrnicon.res)
+  add_executable(mos2nrn ${mswindir}/extra/mos2nrn.cpp)
+  add_executable(nrnbinstr ${mswindir}/extra/nrnbinstr.cpp)
+  install(TARGETS neuron mos2nrn nrnbinstr DESTINATION bin)
+endif()
+
+# =============================================================================
+# Build a Windows setup.exe distribution.
+# =============================================================================
+if(MINGW AND (CMAKE_INSTALL_PREFIX MATCHES "C:/marshalnrn/nrn"))
+  # Presumes previous 'make install' with -DCMAKE_INSTALL_PREFIX="C:/marshalnrn/nrn"
+  # Note: see hard coded set for dos_marshal_dir in ConfigFileSetting.cmake
+  add_custom_target(
+    setup_exe
+    COMMAND bash ${PROJECT_SOURCE_DIR}/mingw_files/nrnmingwenv.sh ${CMAKE_INSTALL_PREFIX} -cmake ${PROJECT_SOURCE_DIR}
+    COMMAND "c:/Program Files (x86)/NSIS/makensisw" ${PROJECT_BINARY_DIR}/src/mswin/nrnsetupmingw.nsi
+    DEPENDS ${PROJECT_SOURCE_DIR}/mingw_files/nrnmingwenv.sh
+    COMMENT "Copy minimal mingw devel files for mknrndll to ${CMAKE_INSTALL_PREFIX}")
+endif()
+if (MINGW)
+  # install mswin specfic
+  set(MSWIN_SRC_DIR ${PROJECT_SOURCE_DIR}/src/mswin)
+  set(MSWIN_FILES
+    ${PROJECT_SOURCE_DIR}/src/parallel/test0.hoc
+    ${PROJECT_SOURCE_DIR}/src/parallel/test0.py
+    ${MSWIN_SRC_DIR}/notes.txt)
+  set(MSWIN_BIN_FILES
```

Comment:
The construction of file lists for libnrniv  was done in cmake/NeuronFileLists.cmake. Should these lists be moved there or a new cmake/SetupExeFileLists.cmake started?

Reply:
I would move it to `cmake/NeuronFileLists.cmake`.